cli/src/cmd/forge/script/mod.rs

```diff
@@ -451,10 +453,50 @@ impl ScriptArgs {
     /// the user.
     fn check_contract_sizes(
         &self,
-        transactions: Option<&VecDeque<TypedTransaction>>,
+        result: &ScriptResult,
         known_contracts: &BTreeMap<ArtifactId, ContractBytecodeSome>,
     ) -> eyre::Result<()> {
-        for (data, to) in transactions.iter().flat_map(|txes| {
+        // (name, &init, &deployed)[]
+        let mut bytecodes: Vec<(String, &[u8], &[u8])> = vec![];
+
+        // From artifacts
+        known_contracts.iter().for_each(|(artifact, bytecode)| {
+            const MSG: &str = "Code should have been linked before.";
+            let init_code = bytecode.bytecode.object.as_bytes().expect(MSG);
+            // Ignore abstract contracts
+            if let Some(ref deployed_code) = bytecode.deployed_bytecode.bytecode {
+                let deployed_code = deployed_code.object.as_bytes().expect(MSG);
+                bytecodes.push((artifact.name.clone(), init_code, deployed_code));
+            }
+        });
+
+        // From traces
+        let create_nodes = result.traces.iter().flat_map(|(_, traces)| {
+            traces
+                .arena
+                .iter()
+                .filter(|node| matches!(node.kind(), CallKind::Create | CallKind::Create2))
+        });
+        let mut unknown_c = 0usize;
+        for node in create_nodes {
+            // Calldata == init code
+            if let RawOrDecodedCall::Raw(ref init_code) = node.trace.data {
+                // Output is the runtime code
+                if let RawOrDecodedReturnData::Raw(ref deployed_code) = node.trace.output {
+                    // Only push if it was not present already
+                    if !bytecodes.iter().any(|(_, b, _)| b == init_code) {
+                        bytecodes.push((format!("Unknown{}", unknown_c), init_code, deployed_code));
+                        unknown_c += 1;
+                    }
+                    continue
+                }
+            }
+            // Both should be raw and not decoded since it's just bytecode
+            unreachable!("Create data was not raw");
```

Comment:
since this function returns a "result", I would rather return this as an error, even if we assume that it is unreachable